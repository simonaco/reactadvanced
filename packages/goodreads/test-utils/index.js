import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { render as rtlRender } from '@testing-library/react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'

const createRenderer = (reducerInitialState, reducer) => (
  ui,
  {
    initialState = reducerInitialState,
    store = createStore(reducer, initialState),
    ...renderOptions
  } = {}
) => {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <Router>{children}</Router>
      </Provider>
    )
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'

// export renderer creator higher order function
export { createRenderer }
