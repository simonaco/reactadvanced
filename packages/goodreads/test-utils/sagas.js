// packages/goodreads/src/test-utils/sagas.js
import { runSaga } from 'redux-saga'

export async function executeSaga(saga, initialAction) {
  const dispatched = []

  await runSaga(
    {
      dispatch: (action) => dispatched.push(action),
    },
    saga,
    initialAction
  ).done

  return dispatched
}
