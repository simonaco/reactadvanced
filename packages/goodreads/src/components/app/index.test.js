import React from 'react'
import { createStore } from 'redux'
import { createRenderer } from '../../../test-utils'
import App from '.'
import reducer from './reducer'

describe('test suite for app component', () => {
  let fakeState
  const render = createRenderer(reducer, fakeState)
  beforeEach(() => {
    fakeState = {
      auth: {
        username: 'anonymous',
      },
      books: {
        isLoading: false,
        meta: [],
        images: [],
        booksInProgress: [],
        ratings: [],
      },
    }
  })
  it('renders app component and matches snapshot', () => {
    const store = createStore(() => fakeState)
    const component = render(<App />, { store })
    expect(component).toMatchSnapshot()
  })
})
