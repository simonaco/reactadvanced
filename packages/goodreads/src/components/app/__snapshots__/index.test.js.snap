// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suite for app component renders app component and matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="css-9suweq"
        >
          <a
            href="/"
          >
            <h1
              class="sc-AxjAm bwqKbb"
              color="#2d8fd5"
              font-family="Artifika"
              font-size="12,12,13,13,13,14"
              font-weight="bold"
            >
              DashBoard
            </h1>
          </a>
          <div
            class="css-1ntkiwo"
          />
          <a
            href="/login"
          >
            <h1
              class="sc-AxjAm bwqKbb"
              color="#2d8fd5"
              font-family="Artifika"
              font-size="12,12,13,13,13,14"
              font-weight="bold"
            >
              Login
            </h1>
          </a>
        </div>
        <main
          style="height: 70vh;"
        >
          <h1
            class="sc-AxjAm jubnsp"
            color="#2d8fd5"
            font-family="Artifika"
            font-size="12,12,13,13,13,14"
            font-weight="700"
          >
            Books
          </h1>
          <div
            class="css-13ieuz9"
          />
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="css-9suweq"
      >
        <a
          href="/"
        >
          <h1
            class="sc-AxjAm bwqKbb"
            color="#2d8fd5"
            font-family="Artifika"
            font-size="12,12,13,13,13,14"
            font-weight="bold"
          >
            DashBoard
          </h1>
        </a>
        <div
          class="css-1ntkiwo"
        />
        <a
          href="/login"
        >
          <h1
            class="sc-AxjAm bwqKbb"
            color="#2d8fd5"
            font-family="Artifika"
            font-size="12,12,13,13,13,14"
            font-weight="bold"
          >
            Login
          </h1>
        </a>
      </div>
      <main
        style="height: 70vh;"
      >
        <h1
          class="sc-AxjAm jubnsp"
          color="#2d8fd5"
          font-family="Artifika"
          font-size="12,12,13,13,13,14"
          font-weight="700"
        >
          Books
        </h1>
        <div
          class="css-13ieuz9"
        />
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
